### PR TITLE
Organize component folders

### LINKED ISSUE #332: Flatten Folder Structure (MAX_PATH)
### Describe the bug

We're hitting an issue with MAX_PATH again for some reason, even after having tested prior fixes if the repo is checked out too deep on a machine:

```
Severity Code Description Project File Line Suppression State Error DEP1000: Cannot copy file "C:\Users\...\source\uwp\Labs-Windows\labs\StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehaviorCustomSample.xaml.cs" to layout "C:\Users\...\source\uwp\Labs-Windows\platforms\CommunityToolkit.Labs.WinAppSdk\bin\x86\Debug\net6.0-windows10.0.19041.0\AppX\SourceAssets/StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehaviorCustomSample.xaml.cs.dat". DirectoryNotFoundException - Could not find a part of the path 'C:\Users\...\source\uwp\Labs-Windows\platforms\CommunityToolkit.Labs.WinAppSdk\bin\x86\Debug\net6.0-windows10.0.19041.0\AppX\SourceAssets/StackedNotificationsBehavior\samples\StackedNotificationsBehavior.Samples\StackedNotificationsBehaviorCustomSample.xaml.cs.dat'. [0x80070003] CommunityToolkit.Labs.WinAppSdk
```

(Incidentally, this is also related to #142, #88, #37 and fixing not needing to copy the assets for Uno.)

Looking at our setup, we duplicate the experiment name in a lot of folders:

image

1) For instance in the samples folder we're already in the `ProjectTemplate\samples` folder, so we should have our subfolders be:
- Content
- Uwp
- Wasm
- WinAppSdk

For tests we could do similar:
- Shared (or maybe Cases?)
- Uwp
- WinAppSdk

-----

2) Alternatively, I like even better just the idea of splitting out the runner projects entirely:
- project template
  - src
  - samples
    - docs/pages just go directly here now
  - tests
    - test pages/cases go directly here now
  - heads (if only we could auto-generate these all on solution opening...)
    - Uwp
    - Wasm
    - WinAppSdk
    - Tests.Uwp
    - Tests.WinAppSdk

Not only does this remove a whole directory layer from our tree, it also makes it immediately clear where everything goes in an experiment and everything that doesn't need to be touched by a contributor and only needed to run the projects is in a single place.

That also makes it easier for us to migrate experiments when we make future infrastructure changes as we should be able to theoretically just wipe and re-create the heads directory of each experiment and not have to reach into so many directories as we would now.

### Steps to reproduce

```text
Clone repo through VS to the standard source folder under users off main...
Build, then try and deploy...
```

### Expected behavior

Should be able to run.

### Screenshots

_No response_

### Code Platform

- [ ] UWP
- [X] WinAppSDK / WinUI 3
- [ ] Web Assembly (WASM)
- [ ] Android
- [ ] iOS
- [ ] MacOS
- [ ] Linux / GTK

### Windows Build Number

- [ ] Windows 10 1809 (Build 17763)
- [ ] Windows 10 1903 (Build 18362)
- [ ] Windows 10 1909 (Build 18363)
- [ ] Windows 10 2004 (Build 19041)
- [ ] Windows 10 20H2 (Build 19042)
- [ ] Windows 10 21H1 (Build 19043)
- [ ] Windows 11 21H2 (Build 22000)
- [ ] Other (specify)

### Other Windows Build number

_No response_

### App minimum and target SDK version

- [ ] Windows 10, version 1809 (Build 17763)
- [ ] Windows 10, version 1903 (Build 18362)
- [ ] Windows 10, version 1909 (Build 18363)
- [ ] Windows 10, version 2004 (Build 19041)
- [ ] Other (specify)

### Other SDK version

_No response_

### Visual Studio Version

_No response_

### Visual Studio Build Number

_No response_

### Device form factor

_No response_

### Additional context

_No response_

### Help us help you

Yes, but only if others can assist.

### PR DESCRIPTION
**Pull request by Arlodotexe**
Prerequisite PR #360 

This PR introduces the following changes:
- [Renamed ./labs to ./components](https://github.com/CommunityToolkit/Labs-Windows/commit/cb6c66b23607a0aa86b9d69afe90cd767346164c)
- [Moved component Sample and Test files up one directory](https://github.com/CommunityToolkit/Labs-Windows/commit/76d2d7b80e21eb3a0fd604d7c1a8384b96248625)
- Closes #332

### CODE REVIEW

File: ReadMe.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ Then you can fork the Labs repo, create a new branch, and start working on your
 ```ascii
 dotnet new --install .\template\ProjectTemplate\
 
-cd labs
+cd components
 
 dotnet new labexp -n MyExperimentNameHere
```
Comment (michael-hawker) [inline ReadMe.md:46]: In the future, we should change the template to be called 'toolkit-component' or something 'tkcmp'? not sure on naming restrictions for shortname on the dotnet templating tool.

File: common/CommunityToolkit.App.Shared/Renderers/ToolkitDocumentationRenderer.xaml.cs [hunk 2]
```diff
@@ -168,17 +168,6 @@ private static async Task<string> GetDocumentationFileContents(ToolkitFrontMatte
 
         var path = metadata.FilePath;
 
-        if (isSingleExperimentHead || isProjectTemplateHead)
```
Comment (michael-hawker) [inline common/CommunityToolkit.App.Shared/Renderers/ToolkitDocumentationRenderer.xaml.cs:171]: Oh did you fix the Uno issue here too? Can we close that issue in our repo?

**Review by michael-hawker — COMMENTED** [review]
Quick pass on GH review. Will try in codespaces in a bit.

File: common/CommunityToolkit.App.Shared/Renderers/ToolkitDocumentationRenderer.xaml.cs (hunk 2 above)
Comment (Arlodotexe) [reply common/CommunityToolkit.App.Shared/Renderers/ToolkitDocumentationRenderer.xaml.cs]: I didn't make this change on this branch. See latest merged commit.

File: ReadMe.md (hunk 1 above)
Comment (Arlodotexe) [reply ReadMe.md]: Sure, if we need to.

**Comment by niels9001** [comment]
Love how more clean this is :)!

**Comment by michael-hawker** [comment]
The all solution is really clean now! 🦙❤

image

**Comment by michael-hawker** [comment]
@Arlodotexe got an error running WASM app-app head locally:

image

File: common/ProjectHeads/App.Head.props [hunk 3]
```diff
@@ -60,10 +60,12 @@
...
 
     <!-- Link/.dat is a workaround for https://github.com/unoplatform/uno/issues/8649 -->
-    <Content Include="$(MSBuildProjectDirectory)\..\..\samples\**\*.cs" Exclude="$(MSBuildProjectDirectory)\..\..\samples\obj\**\*.cs;$(MSBuildProjectDirectory)\..\..\samples\bin\**\*.cs" Link="SourceAssets/%(RecursiveDir)%(FileName)%(Extension).dat" />
+    <Content Include="$(MSBuildProjectDirectory)\..\..\samples\**\*.cs" Exclude="$(MSBuildProjectDirectory)\..\..\samples\obj\**\*.cs;$(MSBuildProjectDirectory)\..\..\samples\bin\**\*.cs">
+      <Link Condition="$(IsWasm) == 'true'">SourceAssets/%(RecursiveDir)%(FileName)%(Extension).dat</Link>
+    </Content>
 
     <!-- Include markdown files from all samples so the head can access them in the source generator -->
-    <AdditionalFiles Include="$(MSBuildProjectDirectory)\..\..\samples\**\*.md" Exclude="$(MSBuildProjectDirectory)\..\..\**\obj\**\*.md;$(MSBuildProjectDirectory)\..\..\**\bin\**\*.md"/>
+    <AdditionalFiles Include="$(MSBuildProjectDirectory)\..\..\samples\*.md" Exclude="$(MSBuildProjectDirectory)\..\..\**\obj\**\*.md;$(MSBuildProjectDirectory)\..\..\**\bin\**\*.md"/>
```
Comment (michael-hawker) [inline common/ProjectHeads/App.Head.props:68]: We need to update uno and unify this anyway... but this means we wouldn't pick up markdown in subfolders which will probably be a thing later. Let's see if we can fix with uno and clean this up after we test sub-modules...

**Review by michael-hawker — APPROVED** [review]

**Pull request merged**